src/mariadb.ts

```diff
@@ -0,0 +1,68 @@
+const mariadb = require("mariadb");
+const uuidv4 = require('uuid/v4');
+
+export class MariaDB {
+    constructor() {
+
+    }
+
+    public async register(username: string, password: string) {
+        const conn = await this.createConnection()
+        const otherUsers = await conn.query(
+            `SELECT * 
+             FROM Users
+             WHERE UserName="${username}"`
+        );
+        if (otherUsers.length > 0) {
+            conn.end();
+            throw new Error('Username is taken');
+        }
+        await conn.query(
+		    `INSERT INTO Users (UserName, Password, Credit) values (
+			    "${username}",
+                "${password}",
+                0
+            )`);
+        conn.end();
+        
+        return await this.loginUser(username, password);
+    }
+
+    public async loginUser(username: string, password: string) {
+        const conn = await this.createConnection();
+        const requestedUser = await conn.query(
+            `SELECT *
+             FROM Users
+             WHERE UserName="${username}"`
+        );
+        if (requestedUser.length == 0) {
+            throw new Error('Invalid Username');
+        }
+        if (requestedUser[0].Password != password) {
+            throw new Error('Invalid Password')
+        }
```

Comment:
Nit: You could combine this as 
```
if (requestedUser.length == 0 || requestedUser[0].Password != password) {
    throw new Error('Invalid Username or Password');
}
```
And then just look for that error message in the other file.